src/components/Visualization/Visualization.js

```diff
@@ -52,14 +56,19 @@ export const Visualization = ({ visualization, onResponseReceived }) => {
     // analytics
     const fetchAnalyticsData = useCallback(async () => {
         const analyticsEngine = Analytics.getAnalytics(dataEngine)
+        const params = { completedOnly: visualization.completedOnly }
+
+        if (relativePeriodDate) {
+            params.relativePeriodDate = relativePeriodDate
+        }
 
         const req = new analyticsEngine.request()
             .fromVisualization(visualization)
             .withProgram(visualization.program.id)
             .withStage(visualization.programStage.id)
             .withDisplayProperty('NAME') // TODO from settings ?!
             .withOutputType(visualization.outputType)
-            .withParameters({ completedOnly: visualization.completedOnly })
+            .withParameters(params)
```

Comment:
There is a specific method for the period date: `withRelativePeriodDate(relativePeriodDate)` although the generic `withParameters` also works.